server/sonar-web/src/main/js/apps/marketplace/App.tsx

```diff
@@ -0,0 +1,209 @@
+import { translate } from '../../helpers/l10n';
+import { filterPlugins, parseQuery, Query, serializeQuery } from './utils';
+
+export interface Props {
+  editionsUrl: string;
+  location: { pathname: string; query: RawQuery };
+  sonarqubeVersion: string;
+  updateCenterActive: boolean;
+}
+
+interface State {
+  editionStatus?: EditionStatus;
+  loading: boolean;
+  pending: {
+    installing: PluginPending[];
+    updating: PluginPending[];
+    removing: PluginPending[];
+  };
+  plugins: Plugin[];
+}
+
+export default class App extends React.PureComponent<Props, State> {
+  mounted: boolean;
+
+  static contextTypes = {
+    router: PropTypes.object.isRequired
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      loading: true,
+      pending: {
+        installing: [],
+        updating: [],
+        removing: []
+      },
+      plugins: []
+    };
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+    this.fetchPendingPlugins();
+    this.fetchEditionStatus();
+    this.fetchQueryPlugins();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.location.query.filter !== this.props.location.query.filter) {
+      this.fetchQueryPlugins();
+    }
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  fetchQueryPlugins = () => {
+    const query = parseQuery(this.props.location.query);
+    if (query.filter === 'updates') {
+      this.fetchUpdatesOnly();
+    } else {
+      this.fetchAllPlugins();
+    }
+  };
+
+  fetchAllPlugins = () => {
+    this.setState({ loading: true });
+    Promise.all([getInstalledPluginsWithUpdates(), getAvailablePlugins()]).then(
+      ([installed, available]) => {
+        if (this.mounted) {
+          this.setState({
+            loading: false,
+            plugins: sortBy(uniqBy([...installed, ...available.plugins], 'key'), 'name')
+          });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loading: false });
+        }
+      }
+    );
+  };
+
+  fetchUpdatesOnly = () => {
+    this.setState({ loading: true });
+    getPluginUpdates().then(
+      plugins => {
+        if (this.mounted) {
+          this.setState({ loading: false, plugins });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loading: false });
+        }
+      }
+    );
+  };
+
+  fetchPendingPlugins = () =>
+    getPendingPlugins().then(
+      pending => {
+        if (this.mounted) {
+          this.setState({ pending });
+        }
+      },
+      () => {}
+    );
+
+  fetchEditionStatus = () =>
+    getEditionStatus().then(
+      editionStatus => {
+        if (this.mounted) {
+          this.updateEditionStatus(editionStatus);
+          setTimeout(this.fetchEditionStatus, 5000);
+        }
+      },
+      () => {}
+    );
+
+  updateEditionStatus = (editionStatus: EditionStatus) =>
+    this.setState({ editionStatus: editionStatus });
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected property shorthand in object literal ('{editionStatus}'). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3498)